synapse/federation/transaction_queue.py

```diff
@@ -239,6 +239,21 @@ def handle_room_events(events):
                     "events", next_token
                 )
 
+                synapse.metrics.event_processing_positions.set(
+                    next_token, "federation_sender",
+                )
+
+                if events:
+                    now = self.clock.time_msec()
+                    ts = yield self.store.get_received_ts(events[-1].event_id)
```

Comment:
seems a shame to have to do a db hit for this, but I guess working around it is a faff.